kernel/src/table.rs

```diff
@@ -80,6 +81,23 @@ impl Table {
         Snapshot::try_new(self.location.clone(), engine, version)
     }
 
+    /// Create a [`TableChanges`] to get a change data feed for the table between `start_version`,
+    /// and `end_version`. If no `end_version` is supplied, the latest version will be used as the
+    /// `end_version`.
+    pub fn table_changes(
```

Comment:
Don't we need to be able to specify:
1.  A schema? Does CDF always return the full table schema? 
2.  A predicate? Can't you get a CDF with a predicate?

Reply:
This is usually done in the [Scan Builder](https://github.com/delta-io/delta-kernel-rs/blob/main/kernel/src/scan/mod.rs#L42). The plan is to specify the schema and predicate when building a `TableChangesScan`.

Reply:
since this is the main 'entrypoint' API for users to interact with reading CDF can we add some docs on semantics and ideally examples too? important things to call out: (1) require that CDF is enabled for the whole range, (2) require the same schema for the whole range (for now!), (3) how do i scan this thing?

(i'm probably forgetting other bits of semantics so don't take the above as exhaustive)